include/morphio/iterators/breadth.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
```

Comment:
If I'm reading this right, you basically just manually templated things - which seems counter-intuitive to me.

I think it would be worthwhile figuring out why the initial code didn't work.

Reply:
I have updated the diagnostic. I'm pretty sure there is a way to solve this without getting rid of the templates but I already spent a lot of time on it and this is blocking the compilation on mac so this is quite a big issue.